### ISSUE TITLE
Potential of over subsumption due to elimination of dynamic memory addresses

### ISSUE THREAD

**Issue by domainexpert** [tracer-x/klee#242]
Given examples such as `basic/malloc3.c` of `klee-examples` repository, over subsumption may be possible since in subsumption check, what is happening is that we only track one level of indirection and higher levels are possibly ignored in subsumption checks. Need to check that this is not the case.

**Comment by domainexpert** [comment]
Higher levels of indirections are no longer ignored in subsumption check. Due to #284, we have complete memory modeling, including all historical updates to the memory, which can still be reachable allocated memory regions, such as nodes in a linked list, which are reachable only from the head variable.

With #284, the memory address in the interpolant table is also identified using its corresponding allocation base address. The variables that are matched first in subsumption checks between the tabled interpolant and the state, have the most recent allocation base address. The others can only be reached via one or more indirections (they are in the historical portion of the interpolant/state), and an attempt to unify them between the states and the interpolant is made at subsumption check.

In summary, the complete information on the memory already gets considered at subsumption checks, no matter how many levels of indirection.
